src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Headers.cs

```diff
@@ -343,18 +343,23 @@ public async Task SendAsync_RequestHeaderInResponse_Success(string name, string
             });
         }
 
-        [OuterLoop("Uses external servers")]
+        [OuterLoop("Uses external servers", PlatformDetection.LocalEchoServerIsNotAvailable)]
         [Theory]
         [InlineData(false)]
         [InlineData(true)]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/53018", TestPlatforms.Browser)]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/37669", TestPlatforms.Browser)]
         public async Task SendAsync_GetWithValidHostHeader_Success(bool withPort)
         {
             if (UseVersion == HttpVersion.Version30)
             {
                 // External servers do not support HTTP3 currently.
                 return;
             }
+            if (PlatformDetection.IsBrowser && withPort)
+            {
+                // we already have custom port with the local echo server
+                return;
+            }
```

Comment:
Could you please explain the reasoning for this?

Reply:
We are passing `localhost:45678` including the port as server address. Which means we could not append another port to it on local echo server.

Reply:
`private static readonly string DefaultHttp2AzureServer = "corefx-net-http2.azurewebsites.net";`

`public static string Http2SecureHost => GetValue("DOTNET_TEST_HTTP2SECUREHOST", DefaultHttp2AzureServer);`

So, `DefaultHttp2AzureServer` doesn't have a port number. But the value that we are setting for `DOTNET_TEST_HTTP2SECUREHOST` with XHarness, does? That doesn't sound right. It changes the tests! We should have the same thing for the value from XHarness too. You could split off the port, when reading the value, if present.

And then this change wouldn't be needed either.

Reply:
After discussion we agreed on adding `Configuration.Http.SecurePort` and comment about the port number in the env variables. Also relevant fix in this test.